tests/trainer/test_checkpoint.py

```diff
@@ -1257,18 +1259,75 @@ def get_trainer(
         eval_dataset = RandomClassificationDataset(size=12)
         train_batch_size = 2
 
-        return Trainer(
-            model=model,
-            train_dataloader=DataLoader(
+        class _DistributedBatchSampler(DistributedSampler):
```

Comment:
should this be outside the helper fn for cleanliness...? Not strongly opinionated, @dakinggg  thoughts?